ONSdigital/rm-common-config
# Fail build on checkstyle violation

# Motivation and Context
We should fail the build when there is a checkstyle to enforce code
being in line with checkstyle. This change adds the severity level to
ensure the build falls over. Also moving the file back to it's original
name to prevent other builds as it depends on the file.

# What has changed
* Rename back to old checkstyle file name
* Make build fail on checkstyle violation

# How to test?
1. Build the service with the latest pom version
1. Go into rm-action-service
1. Bump the parent version in the pom to the recently build version
1. Remove execution block from checkstyle plugin
    ```
                <executions>
                     <execution>
                         <id>test</id>
                         <goals>
                             <goal>check</goal>
                         </goals>
                         <configuration>
                             <configLocation>checkstyle.xml</configLocation>
                             <encoding>UTF-8</encoding>
                             <consoleOutput>true</consoleOutput>
                             <violationSeverity>info</violationSeverity>
                         </configuration>
                     </execution>
                 </executions>
    ```
1. Cause checkstyle issue i.e. line length 100+
1. Build and expect failure

Note. You'll have to temporarily change the `<configLocation>checkstyle.xml</configLocation>` on line 180 in the pom.xml on this branch to `https://raw.githubusercontent.com/ONSdigital/rm-common-config/master/checkstyle/checkstyle.xml` because the config hasn't yet merged

# Links
https://trello.com/c/YUQ48iFw/172-automagically-format-code
